demo/src/main/java/com/wishes/demo/dao/AbstractDAO.java

```diff
@@ -0,0 +1,16 @@
+package com.wishes.demo.dao;
+
+import com.wishes.demo.entity.IEntity;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+
+
+public abstract class AbstractDAO<E extends IEntity> implements DAO<E> {
```

Comment:
DAO classes doesn't required here - repositories will be used as data access logic